{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f2b7d54d",
   "metadata": {},
   "source": [
    "# Printing out metabolite names and their cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e2c31edd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HK1 nan\n",
      "HK2 MDE_TMDS\n",
      "HK3 MDE_TMDS\n",
      "GPI TMDE\n",
      "PFK1\n",
      "ALDOA MDE\n",
      "ALDOB MDS\n",
      "ALDOC MDE\n",
      "GAPDH MDE\n",
      "PGK1 MDE\n",
      "PGM\n",
      "ENO1 MDE\n",
      "ENO2 MDE\n",
      "ENO3 nan\n",
      "PKM MDE\n",
      "LDHA TPDE\n",
      "LDHB TPDS\n",
      "LDHC MDS_TMDE\n",
      "LDHD MDS\n",
      "PDHX TMDS\n",
      "PDHA1 TMDS\n",
      "PDHA2 Not-Background\n",
      "PDHB TMDS\n",
      "MDH2 TMDS\n",
      "CS TMDS\n",
      "ACO2 TMDS\n",
      "IDH2 TMDS\n",
      "IDH3A TMDS\n",
      "IDH3B TMDS\n",
      "IDH3G TMDS\n",
      "OGDH TMDS\n",
      "GLUD1 TMDS\n",
      "GLS TMDS\n",
      "HMOX1 MDE\n",
      "SUCLG2 TPDS\n",
      "SUCLG1 TPDS\n",
      "SUCLA2 TMDS\n",
      "SDHD nan\n",
      "SDHC TMDE\n",
      "SDHB TMDS\n",
      "SDHA TMDS\n",
      "FH TMDS\n",
      "MDH2 TMDS\n",
      "CPS1 nan\n",
      "OTC Not-Background\n",
      "ASS1 TPDS\n",
      "ASL TMDS\n",
      "ARG1 TMDS\n",
      "GPX1 nan\n",
      "GPX2 MDS_TMDE\n",
      "GPX3 TPDS\n",
      "GPX4 TMDS\n",
      "GPX5 Not-Background\n",
      "GPX6 Not-Background\n",
      "GPX7 TMDS\n",
      "GPX8 TPDE_TMDS\n",
      "CTH TPDS\n",
      "CBS Not-Background\n",
      "SHMT1 TPDS\n",
      "SHMT2 MDE_TMDS\n",
      "PSPH TMDS\n",
      "PSAT1 TPDS\n",
      "PHGDH TPDS\n",
      "BHMT TPDS\n",
      "BHMT2 TPDS\n",
      "MAT2B nan\n",
      "MAT2A nan\n",
      "MAT1A MDE_TMDS\n",
      "AHCY TPDS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/sj/4wqsfdtd6093v9746b0t3mq40000gn/T/ipykernel_69647/887256280.py:12: DtypeWarning: Columns (1,2,411,441,442,446,447,702,703,704) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(os.path.join(output_dir, f'sircle_PorMandR_{cancer}.csv'))\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "\n",
    "cancer = 'ClearCellRenalCellCarcinoma-New'\n",
    "input_dir = ''\n",
    "output_dir = ''\n",
    "supp_dir = 'Required_Refs'\n",
    "fig_dir = ''\n",
    "regLabel = 'RG2_Changes_filtered'\n",
    "\n",
    "df = pd.read_csv(os.path.join(output_dir, f'sircle_PorMandR_{cancer}.csv'))\n",
    "\n",
    "# Now just print out the cluster label and then the gene for the different metabolites\n",
    "metabolites = ['HK1', 'HK2', 'HK3', 'GPI', 'PFK1', 'ALDOA', \n",
    "               'ALDOB', 'ALDOC', 'GAPDH', 'PGK1', 'PGM', 'ENO1', 'ENO2', \n",
    "               'ENO3', 'PKM', 'LDHA', 'LDHB', 'LDHC', 'LDHD', 'PDHX', 'PDHA1',\n",
    "               'PDHA2', 'PDHB', 'MDH2', 'CS', 'ACO2', 'IDH2', 'IDH3A', 'IDH3B',\n",
    "               'IDH3G', 'OGDH', 'GLUD1', 'GLS', 'HMOX1', 'SUCLG2', 'SUCLG1', 'SUCLA2', \n",
    "               'SDHD', 'SDHC', 'SDHB', 'SDHA', 'FH', 'MDH2', 'CPS1', 'OTC', 'ASS1', \n",
    "               'ASL', 'ARG1', 'GPX1', 'GPX2', 'GPX3', 'GPX4', 'GPX5', 'GPX6', \n",
    "              'GPX7', 'GPX8', 'CTH', 'CBS', 'SHMT1', 'SHMT2', 'PSPH', 'PSAT1',\n",
    "              'PHGDH', 'BHMT', 'BHMT2', 'MAT2B', 'MAT2A', 'MAT1A', 'AHCY']\n",
    "\n",
    "for m in metabolites:\n",
    "    try:\n",
    "        g = df[df['gene_name'] == m]['RG2_Changes_filtered'].values[0]\n",
    "        print(m, g)\n",
    "    except:\n",
    "        print(m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8b6170c6-5ae0-4686-968f-f2a611c374d6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/sj/4wqsfdtd6093v9746b0t3mq40000gn/T/ipykernel_69647/2647856554.py:5: DtypeWarning: Columns (1,2,353,381,383,384,388,389,583,584,585) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df_ccrcc = pd.read_csv(f'{output_dir}sircle_PorMandR_ClearCellRenalCellCarcinoma.csv')\n",
      "/var/folders/sj/4wqsfdtd6093v9746b0t3mq40000gn/T/ipykernel_69647/2647856554.py:6: DtypeWarning: Columns (1,2,411,441,442,446,447,702,703,704) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df_new = pd.read_csv(f'{output_dir}sircle_PorMandR_ClearCellRenalCellCarcinoma-New.csv')\n",
      "/var/folders/sj/4wqsfdtd6093v9746b0t3mq40000gn/T/ipykernel_69647/2647856554.py:7: DtypeWarning: Columns (1,2,1161,1163,1164,1167,1176,1177,1178,1179,1180,1181,1182,1195,1197,1198,1202,1203,1208,1974,1975,1976) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df_pancan = pd.read_csv(f'{output_dir}sircle_PorMandR_PanCan.csv')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HK1 nan nan nan\n",
      "HK2 MDE MDE_TMDS MDE\n",
      "HK3 MDE MDE_TMDS nan\n",
      "GPI TMDE TMDE nan\n",
      "PFK1 None None None\n",
      "ALDOA MDE MDE nan\n",
      "ALDOB TPDS MDS TPDS_TMDE\n",
      "ALDOC MDE MDE nan\n",
      "GAPDH TPDE MDE MDE_TMDS\n",
      "PGK1 MDE MDE nan\n",
      "PGM None None None\n",
      "ENO1 MDE MDE nan\n",
      "ENO2 MDE MDE TPDE_TMDS\n",
      "ENO3 TMDE nan nan\n",
      "PKM MDE MDE nan\n",
      "LDHA TPDE TPDE MDE_TMDS\n",
      "LDHB TPDS TPDS nan\n",
      "LDHC MDS_TMDE MDS_TMDE Not-Background\n",
      "LDHD MDS MDS TMDS\n",
      "PDHX TMDS TMDS nan\n",
      "PDHA1 TPDS TMDS nan\n",
      "PDHA2 TMDS Not-Background Not-Background\n",
      "PDHB TMDS TMDS nan\n",
      "MDH2 TMDS TMDS nan\n",
      "CS TMDS TMDS nan\n",
      "ACO2 TPDS TMDS nan\n",
      "IDH2 TMDS TMDS nan\n",
      "IDH3A TMDS TMDS nan\n",
      "IDH3B TMDS TMDS nan\n",
      "IDH3G TMDS TMDS nan\n",
      "OGDH TPDS TMDS nan\n",
      "GLUD1 TMDS TMDS nan\n",
      "GLS TMDS TMDS nan\n",
      "HMOX1 MDE MDE nan\n",
      "SUCLG2 MDS TPDS nan\n",
      "SUCLG1 TPDS TPDS nan\n",
      "SUCLA2 TMDS TMDS nan\n",
      "SDHD nan nan nan\n",
      "SDHC TMDS TMDE nan\n",
      "SDHB TMDS TMDS nan\n",
      "SDHA TMDS TMDS nan\n",
      "FH TMDS TMDS nan\n",
      "MDH2 TMDS TMDS nan\n",
      "CPS1 TMDS nan nan\n",
      "OTC Not-Background Not-Background Not-Background\n",
      "ASS1 TPDS TPDS nan\n",
      "ASL TMDS TMDS nan\n",
      "ARG1 TMDE TMDS TPDS_TMDE\n",
      "GPX1 nan nan nan\n",
      "GPX2 nan MDS_TMDE MDE_TMDS\n",
      "GPX3 MDS TPDS MDS\n",
      "GPX4 nan TMDS nan\n",
      "GPX5 Not-Background Not-Background Not-Background\n",
      "GPX6 Not-Background Not-Background Not-Background\n",
      "GPX7 TMDE TMDS TMDE\n",
      "GPX8 TPDE TPDE_TMDS TMDE\n",
      "CTH TPDS TPDS nan\n",
      "CBS Not-Background Not-Background Not-Background\n",
      "SHMT1 TPDS TPDS nan\n",
      "SHMT2 MDE_TMDS MDE_TMDS nan\n",
      "PSPH nan TMDS nan\n",
      "PSAT1 TPDS TPDS TPDE_TMDS\n",
      "PHGDH MDS TPDS nan\n",
      "BHMT TPDS TPDS nan\n",
      "BHMT2 TPDS TPDS Not-Background\n",
      "MAT2B nan nan nan\n",
      "MAT2A nan nan nan\n",
      "MAT1A nan MDE_TMDS nan\n",
      "AHCY TPDS TPDS nan\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "\n",
    "df_ccrcc = pd.read_csv(f'{output_dir}sircle_PorMandR_ClearCellRenalCellCarcinoma.csv')\n",
    "df_new = pd.read_csv(f'{output_dir}sircle_PorMandR_ClearCellRenalCellCarcinoma-New.csv')\n",
    "df_pancan = pd.read_csv(f'{output_dir}sircle_PorMandR_PanCan.csv')\n",
    "# Now just print out the cluster label and then the gene for the different metabolites\n",
    "metabolites = ['HK1', 'HK2', 'HK3', 'GPI', 'PFK1', 'ALDOA', \n",
    "               'ALDOB', 'ALDOC', 'GAPDH', 'PGK1', 'PGM', 'ENO1', 'ENO2', \n",
    "               'ENO3', 'PKM', 'LDHA', 'LDHB', 'LDHC', 'LDHD', 'PDHX', 'PDHA1',\n",
    "               'PDHA2', 'PDHB', 'MDH2', 'CS', 'ACO2', 'IDH2', 'IDH3A', 'IDH3B',\n",
    "               'IDH3G', 'OGDH', 'GLUD1', 'GLS', 'HMOX1', 'SUCLG2', 'SUCLG1', 'SUCLA2', \n",
    "               'SDHD', 'SDHC', 'SDHB', 'SDHA', 'FH', 'MDH2', 'CPS1', 'OTC', 'ASS1', \n",
    "               'ASL', 'ARG1', 'GPX1', 'GPX2', 'GPX3', 'GPX4', 'GPX5', 'GPX6', \n",
    "              'GPX7', 'GPX8', 'CTH', 'CBS', 'SHMT1', 'SHMT2', 'PSPH', 'PSAT1',\n",
    "              'PHGDH', 'BHMT', 'BHMT2', 'MAT2B', 'MAT2A', 'MAT1A', 'AHCY']\n",
    "\n",
    "rows = []\n",
    "same_new = 0\n",
    "same_pancan = 0\n",
    "for m in metabolites:\n",
    "    g_ccrcc, g_pancan, g_new = None, None, None\n",
    "    try:\n",
    "        g_ccrcc = df_ccrcc[df_ccrcc['gene_name'] == m]['RG2_Changes_filtered'].values[0]\n",
    "    except:\n",
    "        x = ''\n",
    "    try:\n",
    "        g_pancan = df_pancan[df_pancan['gene_name'] == m]['RG2_Changes_filtered'].values[0]\n",
    "    except:\n",
    "        x = ''\n",
    "    try:\n",
    "        g_new = df_new[df_new['gene_name'] == m]['RG2_Changes_filtered'].values[0]\n",
    "    except:\n",
    "        x = ''\n",
    "    if g_ccrcc:\n",
    "        if g_ccrcc == g_new:\n",
    "            same_new +=1\n",
    "        if g_ccrcc == g_pancan:\n",
    "            same_pancan += 1\n",
    "    print(m, g_ccrcc, g_new, g_pancan)\n",
    "    rows.append([m, g_ccrcc, g_new, g_pancan])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e892714-2d49-4904-93e8-616bb2348700",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "06f54360-5ca0-4865-a526-c8425e4ddb59",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>metabolite</th>\n",
       "      <th>ccRCC</th>\n",
       "      <th>ccRCC-ITH</th>\n",
       "      <th>PanCan</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HK1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HK2</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE_TMDS</td>\n",
       "      <td>MDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HK3</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE_TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GPI</td>\n",
       "      <td>TMDE</td>\n",
       "      <td>TMDE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PFK1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>BHMT2</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>Not-Background</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>MAT2B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>MAT2A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>MAT1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MDE_TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>AHCY</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>69 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   metabolite ccRCC ccRCC-ITH          PanCan\n",
       "0         HK1   NaN       NaN             NaN\n",
       "1         HK2   MDE  MDE_TMDS             MDE\n",
       "2         HK3   MDE  MDE_TMDS             NaN\n",
       "3         GPI  TMDE      TMDE             NaN\n",
       "4        PFK1  None      None            None\n",
       "..        ...   ...       ...             ...\n",
       "64      BHMT2  TPDS      TPDS  Not-Background\n",
       "65      MAT2B   NaN       NaN             NaN\n",
       "66      MAT2A   NaN       NaN             NaN\n",
       "67      MAT1A   NaN  MDE_TMDS             NaN\n",
       "68       AHCY  TPDS      TPDS             NaN\n",
       "\n",
       "[69 rows x 4 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(rows, columns=['metabolite', 'ccRCC', 'ccRCC-ITH', 'PanCan'])\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "9d11e091-874a-4db2-95c1-9c7fec62eb35",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>metabolite</th>\n",
       "      <th>ccRCC</th>\n",
       "      <th>ccRCC-ITH</th>\n",
       "      <th>PanCan</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HK2</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE_TMDS</td>\n",
       "      <td>MDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HK3</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE_TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GPI</td>\n",
       "      <td>TMDE</td>\n",
       "      <td>TMDE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ALDOA</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ALDOB</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>MDS</td>\n",
       "      <td>TPDS_TMDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ALDOC</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GAPDH</td>\n",
       "      <td>TPDE</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE_TMDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PGK1</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENO1</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENO2</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE</td>\n",
       "      <td>TPDE_TMDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENO3</td>\n",
       "      <td>TMDE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PKM</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>LDHA</td>\n",
       "      <td>TPDE</td>\n",
       "      <td>TPDE</td>\n",
       "      <td>MDE_TMDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>LDHB</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>LDHC</td>\n",
       "      <td>MDS_TMDE</td>\n",
       "      <td>MDS_TMDE</td>\n",
       "      <td>Not-Background</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>LDHD</td>\n",
       "      <td>MDS</td>\n",
       "      <td>MDS</td>\n",
       "      <td>TMDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PDHX</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PDHA1</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PDHA2</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>Not-Background</td>\n",
       "      <td>Not-Background</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PDHB</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MDH2</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>CS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ACO2</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>IDH2</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>IDH3A</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>IDH3B</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>IDH3G</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>OGDH</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GLUD1</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GLS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>HMOX1</td>\n",
       "      <td>MDE</td>\n",
       "      <td>MDE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>SUCLG2</td>\n",
       "      <td>MDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>SUCLG1</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>SUCLA2</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>SDHC</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>SDHB</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>SDHA</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>FH</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MDH2</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>CPS1</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>OTC</td>\n",
       "      <td>Not-Background</td>\n",
       "      <td>Not-Background</td>\n",
       "      <td>Not-Background</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>ASS1</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>ASL</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>ARG1</td>\n",
       "      <td>TMDE</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TPDS_TMDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>GPX3</td>\n",
       "      <td>MDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>MDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>GPX5</td>\n",
       "      <td>Not-Background</td>\n",
       "      <td>Not-Background</td>\n",
       "      <td>Not-Background</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>GPX6</td>\n",
       "      <td>Not-Background</td>\n",
       "      <td>Not-Background</td>\n",
       "      <td>Not-Background</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>GPX7</td>\n",
       "      <td>TMDE</td>\n",
       "      <td>TMDS</td>\n",
       "      <td>TMDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>GPX8</td>\n",
       "      <td>TPDE</td>\n",
       "      <td>TPDE_TMDS</td>\n",
       "      <td>TMDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>CTH</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>CBS</td>\n",
       "      <td>Not-Background</td>\n",
       "      <td>Not-Background</td>\n",
       "      <td>Not-Background</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>SHMT1</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>SHMT2</td>\n",
       "      <td>MDE_TMDS</td>\n",
       "      <td>MDE_TMDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>PSAT1</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDE_TMDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>PHGDH</td>\n",
       "      <td>MDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>BHMT</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>BHMT2</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>Not-Background</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>AHCY</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>TPDS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   metabolite           ccRCC       ccRCC-ITH          PanCan\n",
       "1         HK2             MDE        MDE_TMDS             MDE\n",
       "2         HK3             MDE        MDE_TMDS             NaN\n",
       "3         GPI            TMDE            TMDE             NaN\n",
       "5       ALDOA             MDE             MDE             NaN\n",
       "6       ALDOB            TPDS             MDS       TPDS_TMDE\n",
       "7       ALDOC             MDE             MDE             NaN\n",
       "8       GAPDH            TPDE             MDE        MDE_TMDS\n",
       "9        PGK1             MDE             MDE             NaN\n",
       "11       ENO1             MDE             MDE             NaN\n",
       "12       ENO2             MDE             MDE       TPDE_TMDS\n",
       "13       ENO3            TMDE             NaN             NaN\n",
       "14        PKM             MDE             MDE             NaN\n",
       "15       LDHA            TPDE            TPDE        MDE_TMDS\n",
       "16       LDHB            TPDS            TPDS             NaN\n",
       "17       LDHC        MDS_TMDE        MDS_TMDE  Not-Background\n",
       "18       LDHD             MDS             MDS            TMDS\n",
       "19       PDHX            TMDS            TMDS             NaN\n",
       "20      PDHA1            TPDS            TMDS             NaN\n",
       "21      PDHA2            TMDS  Not-Background  Not-Background\n",
       "22       PDHB            TMDS            TMDS             NaN\n",
       "23       MDH2            TMDS            TMDS             NaN\n",
       "24         CS            TMDS            TMDS             NaN\n",
       "25       ACO2            TPDS            TMDS             NaN\n",
       "26       IDH2            TMDS            TMDS             NaN\n",
       "27      IDH3A            TMDS            TMDS             NaN\n",
       "28      IDH3B            TMDS            TMDS             NaN\n",
       "29      IDH3G            TMDS            TMDS             NaN\n",
       "30       OGDH            TPDS            TMDS             NaN\n",
       "31      GLUD1            TMDS            TMDS             NaN\n",
       "32        GLS            TMDS            TMDS             NaN\n",
       "33      HMOX1             MDE             MDE             NaN\n",
       "34     SUCLG2             MDS            TPDS             NaN\n",
       "35     SUCLG1            TPDS            TPDS             NaN\n",
       "36     SUCLA2            TMDS            TMDS             NaN\n",
       "38       SDHC            TMDS            TMDE             NaN\n",
       "39       SDHB            TMDS            TMDS             NaN\n",
       "40       SDHA            TMDS            TMDS             NaN\n",
       "41         FH            TMDS            TMDS             NaN\n",
       "42       MDH2            TMDS            TMDS             NaN\n",
       "43       CPS1            TMDS             NaN             NaN\n",
       "44        OTC  Not-Background  Not-Background  Not-Background\n",
       "45       ASS1            TPDS            TPDS             NaN\n",
       "46        ASL            TMDS            TMDS             NaN\n",
       "47       ARG1            TMDE            TMDS       TPDS_TMDE\n",
       "50       GPX3             MDS            TPDS             MDS\n",
       "52       GPX5  Not-Background  Not-Background  Not-Background\n",
       "53       GPX6  Not-Background  Not-Background  Not-Background\n",
       "54       GPX7            TMDE            TMDS            TMDE\n",
       "55       GPX8            TPDE       TPDE_TMDS            TMDE\n",
       "56        CTH            TPDS            TPDS             NaN\n",
       "57        CBS  Not-Background  Not-Background  Not-Background\n",
       "58      SHMT1            TPDS            TPDS             NaN\n",
       "59      SHMT2        MDE_TMDS        MDE_TMDS             NaN\n",
       "61      PSAT1            TPDS            TPDS       TPDE_TMDS\n",
       "62      PHGDH             MDS            TPDS             NaN\n",
       "63       BHMT            TPDS            TPDS             NaN\n",
       "64      BHMT2            TPDS            TPDS  Not-Background\n",
       "68       AHCY            TPDS            TPDS             NaN"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.dropna(subset=['ccRCC'])\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "52db2c8b-f292-4b97-b310-ae9d584bf135",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.7068965517241379\n"
     ]
    }
   ],
   "source": [
    "print(same_new/len(df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "59a6e2c8-9405-4dc2-aaae-820c131d6688",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.1206896551724138\n"
     ]
    }
   ],
   "source": [
    "print(same_pancan/len(df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "97a9313a-ddcb-4f46-9a93-d2e5c0e62faa",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df.to_csv('overlap_metabolites.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cdd151e9-1dd6-4ad7-b74f-d6ebfdf91283",
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib_venn import venn3\n",
    "\n",
    "venn3([set(p_go), set(r_go), set(m_go)], set_labels=['Proteomics', 'RNAseq', 'DNA Methylation'],\n",
    "      set_colors=('#0101d7', '#e68e25', '#6aaf44'),)\n",
    "plt.rcParams['svg.fonttype'] = 'none'\n",
    "plt.title('Down')\n",
    "plt.savefig(os.path.join(fig_dir, 'Figure4B_DownGenes_EarlyLate.svg'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3990e842",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/gq/6ljhmvm1713fykdjqbl188pm0000gn/T/ipykernel_92962/1884489283.py:12: DtypeWarning: Columns (1,2,353,381,383,384,388,389,583,584,585) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(os.path.join(output_dir, f'sircle_PorMandR_{cancer}.csv'))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HK2 MDE\n",
      "HK3 MDE\n",
      "ALDOA MDE\n",
      "ALDOC MDE\n",
      "GAPDH TPDE\n",
      "PGK1 MDE\n",
      "ENO1 MDE\n",
      "ENO2 MDE\n",
      "PKM MDE\n",
      "LDHA TPDE\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "\n",
    "cancer = 'ClearCellRenalCellCarcinoma'\n",
    "input_dir = 'Input_RCM'\n",
    "output_dir = 'Output_Data'\n",
    "supp_dir = 'Required_Refs'\n",
    "fig_dir = 'Output_Figures'\n",
    "regLabel = 'RG2_Changes_filtered'\n",
    "\n",
    "df = pd.read_csv(os.path.join(output_dir, f'sircle_PorMandR_{cancer}.csv'))\n",
    "\n",
    "# Now just print out the cluster label and then the gene for the different metabolites\n",
    "metabolites = ['HK2', 'HK3', 'ALDOA', 'ALDOC', 'GAPDH', 'PGK1', 'ENO1', \n",
    "               'ENO2', 'PKM', 'LDHA']\n",
    "\n",
    "for m in metabolites:\n",
    "    try:\n",
    "        g = df[df['gene_name'] == m]['RG2_Changes_filtered'].values[0]\n",
    "        print(m, g)\n",
    "    except:\n",
    "        print(m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2fca22a1-3bc2-44f9-a1c5-78cecf58d042",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HK2 MDE_TMDS\n",
      "HK3 MDE_TMDS\n",
      "ALDOA MDE\n",
      "ALDOC MDE\n",
      "GAPDH MDE\n",
      "PGK1 MDE\n",
      "ENO1 MDE\n",
      "ENO2 MDE\n",
      "PKM MDE\n",
      "LDHA TPDE\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/sj/4wqsfdtd6093v9746b0t3mq40000gn/T/ipykernel_69647/3712534443.py:4: DtypeWarning: Columns (1,2,411,441,442,446,447,702,703,704) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(os.path.join(output_dir, f'sircle_PorMandR_{cancer}.csv'))\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "## NEw ones!\n",
    "df = pd.read_csv(os.path.join(output_dir, f'sircle_PorMandR_{cancer}.csv'))\n",
    "\n",
    "# Now just print out the cluster label and then the gene for the different metabolites\n",
    "metabolites = ['HK2', 'HK3', 'ALDOA', 'ALDOC', 'GAPDH', 'PGK1', 'ENO1', \n",
    "               'ENO2', 'PKM', 'LDHA']\n",
    "\n",
    "for m in metabolites:\n",
    "    try:\n",
    "        g = df[df['gene_name'] == m]['RG2_Changes_filtered'].values[0]\n",
    "        print(m, g)\n",
    "    except:\n",
    "        print(m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "cd3e4318-0540-43bc-b988-5b3846af3c87",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/sj/4wqsfdtd6093v9746b0t3mq40000gn/T/ipykernel_69647/3019080726.py:4: DtypeWarning: Columns (1,2,1161,1163,1164,1167,1176,1177,1178,1179,1180,1181,1182,1195,1197,1198,1202,1203,1208,1974,1975,1976) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f'Part3_Comparison/Input_Data/sircle_PorMandR_PanCan.csv')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HK2 MDE\n",
      "HK3 nan\n",
      "ALDOA nan\n",
      "ALDOC nan\n",
      "GAPDH MDE_TMDS\n",
      "PGK1 nan\n",
      "ENO1 nan\n",
      "ENO2 TPDE_TMDS\n",
      "PKM nan\n",
      "LDHA MDE_TMDS\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "## NEw ones!\n",
    "output_dir = 'Part3_Comparison/Input_Data/'\n",
    "df = pd.read_csv(f'{output_dir}sircle_PorMandR_PanCan.csv')\n",
    "\n",
    "# Now just print out the cluster label and then the gene for the different metabolites\n",
    "metabolites = ['HK2', 'HK3', 'ALDOA', 'ALDOC', 'GAPDH', 'PGK1', 'ENO1', \n",
    "               'ENO2', 'PKM', 'LDHA']\n",
    "\n",
    "for m in metabolites:\n",
    "    try:\n",
    "        g = df[df['gene_name'] == m]['RG2_Changes_filtered'].values[0]\n",
    "        print(m, g)\n",
    "    except:\n",
    "        print(m)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
