{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "41edd770",
   "metadata": {},
   "source": [
    "# Consolidate pathways so that we can do the metabolomics vis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a04b9562",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                   Stage_IV_vs_Stage_I_All_GSEA_Pathways.csv\t                   \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "                                               pathway      pval      padj  \\\n",
      "6               KEGG_ASCORBATE_AND_ALDARATE_METABOLISM  0.027883  0.232360   \n",
      "10                   KEGG_STEROID_HORMONE_BIOSYNTHESIS  0.000698  0.036642   \n",
      "11                      KEGG_OXIDATIVE_PHOSPHORYLATION  0.000826  0.036642   \n",
      "57           KEGG_PORPHYRIN_AND_CHLOROPHYLL_METABOLISM  0.019941  0.194614   \n",
      "62   KEGG_METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450  0.005635  0.089641   \n",
      "64                  KEGG_DRUG_METABOLISM_OTHER_ENZYMES  0.008004  0.116732   \n",
      "83                                     KEGG_CELL_CYCLE  0.000264  0.036642   \n",
      "84                                 KEGG_OOCYTE_MEIOSIS  0.001516  0.053052   \n",
      "85                          KEGG_P53_SIGNALING_PATHWAY  0.012213  0.153762   \n",
      "104                  KEGG_CELL_ADHESION_MOLECULES_CAMS  0.014420  0.157719   \n",
      "109           KEGG_ANTIGEN_PROCESSING_AND_PRESENTATION  0.000838  0.036642   \n",
      "132       KEGG_PROGESTERONE_MEDIATED_OOCYTE_MATURATION  0.004234  0.085325   \n",
      "134               KEGG_ADIPOCYTOKINE_SIGNALING_PATHWAY  0.002113  0.061641   \n",
      "138     KEGG_ALDOSTERONE_REGULATED_SODIUM_REABSORPTION  0.002983  0.074584   \n",
      "140       KEGG_PROXIMAL_TUBULE_BICARBONATE_RECLAMATION  0.013180  0.153762   \n",
      "141                            KEGG_ALZHEIMERS_DISEASE  0.004876  0.085325   \n",
      "142                            KEGG_PARKINSONS_DISEASE  0.027484  0.232360   \n",
      "144                           KEGG_HUNTINGTONS_DISEASE  0.023250  0.214147   \n",
      "165                                        KEGG_ASTHMA  0.004829  0.085325   \n",
      "168                           KEGG_ALLOGRAFT_REJECTION  0.012448  0.153762   \n",
      "174                             KEGG_VIRAL_MYOCARDITIS  0.020017  0.194614   \n",
      "\n",
      "           ES       NES  nMoreExtreme  size  leadingEdge  \n",
      "6    0.587537  1.540481           190    16          NaN  \n",
      "10   0.654090  1.892871             4    24          NaN  \n",
      "11   0.462153  1.704033             6    90          NaN  \n",
      "57   0.548087  1.568666           141    23          NaN  \n",
      "62   0.532872  1.658889            41    34          NaN  \n",
      "64   0.577942  1.654113            56    23          NaN  \n",
      "83   0.602546  1.928077             1    39          NaN  \n",
      "84   0.603294  1.788716            10    27          NaN  \n",
      "85   0.525849  1.596041            89    30          NaN  \n",
      "104 -0.318449 -1.422493            22    85          NaN  \n",
      "109 -0.472242 -1.825362             1    41          NaN  \n",
      "132  0.659466  1.729074            28    16          NaN  \n",
      "134 -0.552940 -1.874310             5    24          NaN  \n",
      "138  0.635651  1.763522            20    20          NaN  \n",
      "140 -0.544339 -1.703191            39    18          NaN  \n",
      "141  0.424413  1.554985            40    86          NaN  \n",
      "142  0.391843  1.421975           228    80          NaN  \n",
      "144  0.394850  1.434857           193    81          NaN  \n",
      "165 -0.542890 -1.790142            13    22          NaN  \n",
      "168 -0.447943 -1.622105            32    31          NaN  \n",
      "174 -0.382627 -1.502292            45    44          NaN  \n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                   Late_vs_Early_TPDE_TMDS_GSEA_Pathways.csv\t                   \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "                                         pathway      pval      padj  \\\n",
      "6   KEGG_NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION  0.003292  0.041415   \n",
      "8                            KEGG_OOCYTE_MEIOSIS  0.003451  0.041415   \n",
      "16               KEGG_JAK_STAT_SIGNALING_PATHWAY  0.024649  0.197190   \n",
      "\n",
      "          ES       NES  nMoreExtreme  size  leadingEdge  \n",
      "6  -0.658415 -1.894727            11    17          NaN  \n",
      "8   0.878130  1.636022            19     5          NaN  \n",
      "16 -0.846145 -1.549914           106     4          NaN  \n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                     Late_vs_Early_TMDE_GSEA_Pathways.csv\t                      \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "                                   pathway      pval      padj        ES  \\\n",
      "15  KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY  0.016882  0.219471 -0.830475   \n",
      "16  KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY  0.016882  0.219471 -0.830475   \n",
      "\n",
      "         NES  nMoreExtreme  size  leadingEdge  \n",
      "15 -1.682063            59     4          NaN  \n",
      "16 -1.682063            59     4          NaN  \n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                   Late_vs_Early_TPDS_TMDE_GSEA_Pathways.csv\t                   \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "                                              pathway      pval      padj  \\\n",
      "1              KEGG_ASCORBATE_AND_ALDARATE_METABOLISM  0.041644  0.183234   \n",
      "2                   KEGG_STEROID_HORMONE_BIOSYNTHESIS  0.016535  0.121253   \n",
      "3                  KEGG_STARCH_AND_SUCROSE_METABOLISM  0.041644  0.183234   \n",
      "4                             KEGG_RETINOL_METABOLISM  0.005062  0.054677   \n",
      "5           KEGG_PORPHYRIN_AND_CHLOROPHYLL_METABOLISM  0.025764  0.161945   \n",
      "6   KEGG_METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450  0.004978  0.054677   \n",
      "7                KEGG_DRUG_METABOLISM_CYTOCHROME_P450  0.006213  0.054677   \n",
      "14        KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION  0.002990  0.054677   \n",
      "15                   KEGG_CHEMOKINE_SIGNALING_PATHWAY  0.057613  0.230453   \n",
      "24                                 KEGG_AXON_GUIDANCE  0.032810  0.180455   \n",
      "31                    KEGG_HEMATOPOIETIC_CELL_LINEAGE  0.002692  0.054677   \n",
      "\n",
      "          ES       NES  nMoreExtreme  size  leadingEdge  \n",
      "1   0.802752  1.459200           230     5          NaN  \n",
      "2   0.728244  1.604836            97    10          NaN  \n",
      "3   0.802752  1.459200           230     5          NaN  \n",
      "4   0.772047  1.701366            29    10          NaN  \n",
      "5   0.795713  1.519386           144     6          NaN  \n",
      "6   0.730575  1.730148            29    13          NaN  \n",
      "7   0.752845  1.701978            36    11          NaN  \n",
      "14 -0.734764 -1.819878            11    12          NaN  \n",
      "15 -0.740379 -1.497316           251     6          NaN  \n",
      "24 -0.747647 -1.584614           139     7          NaN  \n",
      "31 -0.908226 -1.732726            11     5          NaN  \n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                   Late_vs_Early_MDE_TMDS_GSEA_Pathways.csv\t                    \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "                                           pathway      pval      padj  \\\n",
      "0                KEGG_STEROID_HORMONE_BIOSYNTHESIS  0.006767  0.106913   \n",
      "3               KEGG_DRUG_METABOLISM_OTHER_ENZYMES  0.000549  0.043343   \n",
      "13                                 KEGG_CELL_CYCLE  0.021322  0.240629   \n",
      "14                             KEGG_OOCYTE_MEIOSIS  0.006439  0.106913   \n",
      "22                    KEGG_NOTCH_SIGNALING_PATHWAY  0.025116  0.248017   \n",
      "28               KEGG_CELL_ADHESION_MOLECULES_CAMS  0.005335  0.106913   \n",
      "39  KEGG_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY  0.015328  0.201825   \n",
      "40          KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY  0.005542  0.106913   \n",
      "\n",
      "          ES       NES  nMoreExtreme  size  leadingEdge  \n",
      "0   0.875150  1.639698            36     4          NaN  \n",
      "3   0.943345  1.767470             2     4          NaN  \n",
      "13  0.661857  1.598604           120     9          NaN  \n",
      "14  0.759500  1.696611            35     7          NaN  \n",
      "22 -0.776874 -1.640029           113     5          NaN  \n",
      "28 -0.517999 -1.747505            21    23          NaN  \n",
      "39 -0.512668 -1.653863            62    20          NaN  \n",
      "40 -0.583989 -1.804562            22    17          NaN  \n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                      Late_vs_Early_All_GSEA_Pathways.csv\t                      \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "                                               pathway      pval      padj  \\\n",
      "1                         KEGG_CITRATE_CYCLE_TCA_CYCLE  0.005114  0.055938   \n",
      "6               KEGG_ASCORBATE_AND_ALDARATE_METABOLISM  0.013492  0.109754   \n",
      "7                           KEGG_FATTY_ACID_METABOLISM  0.060782  0.247370   \n",
      "10                   KEGG_STEROID_HORMONE_BIOSYNTHESIS  0.007009  0.068143   \n",
      "11                      KEGG_OXIDATIVE_PHOSPHORYLATION  0.000132  0.006244   \n",
      "14     KEGG_ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM  0.000602  0.009572   \n",
      "17      KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION  0.000435  0.008468   \n",
      "19                             KEGG_LYSINE_DEGRADATION  0.008990  0.082801   \n",
      "20                KEGG_ARGININE_AND_PROLINE_METABOLISM  0.004742  0.055938   \n",
      "24                          KEGG_TRYPTOPHAN_METABOLISM  0.000598  0.009572   \n",
      "35   KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SU...  0.019814  0.144478   \n",
      "48        KEGG_GLYOXYLATE_AND_DICARBOXYLATE_METABOLISM  0.051718  0.220750   \n",
      "49                          KEGG_PROPANOATE_METABOLISM  0.018201  0.138486   \n",
      "50                           KEGG_BUTANOATE_METABOLISM  0.037787  0.204882   \n",
      "54              KEGG_PANTOTHENATE_AND_COA_BIOSYNTHESIS  0.023732  0.148322   \n",
      "57           KEGG_PORPHYRIN_AND_CHLOROPHYLL_METABOLISM  0.006882  0.068143   \n",
      "61                    KEGG_AMINOACYL_TRNA_BIOSYNTHESIS  0.042222  0.204882   \n",
      "64                  KEGG_DRUG_METABOLISM_OTHER_ENZYMES  0.003059  0.041173   \n",
      "67                                       KEGG_RIBOSOME  0.043937  0.204882   \n",
      "69                                KEGG_DNA_REPLICATION  0.004941  0.055938   \n",
      "72                           KEGG_BASE_EXCISION_REPAIR  0.030388  0.177264   \n",
      "76                         KEGG_MAPK_SIGNALING_PATHWAY  0.041762  0.204882   \n",
      "79         KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION  0.039503  0.204882   \n",
      "82        KEGG_NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION  0.000403  0.008468   \n",
      "83                                     KEGG_CELL_CYCLE  0.055257  0.230236   \n",
      "84                                 KEGG_OOCYTE_MEIOSIS  0.023277  0.148322   \n",
      "85                          KEGG_P53_SIGNALING_PATHWAY  0.023242  0.148322   \n",
      "88                                       KEGG_LYSOSOME  0.033134  0.187048   \n",
      "90                                     KEGG_PEROXISOME  0.000143  0.006244   \n",
      "93                     KEGG_CARDIAC_MUSCLE_CONTRACTION  0.045659  0.204882   \n",
      "99                     KEGG_TGF_BETA_SIGNALING_PATHWAY  0.012332  0.107904   \n",
      "102                                KEGG_FOCAL_ADHESION  0.000379  0.008468   \n",
      "103                      KEGG_ECM_RECEPTOR_INTERACTION  0.000329  0.008468   \n",
      "104                  KEGG_CELL_ADHESION_MOLECULES_CAMS  0.001606  0.023427   \n",
      "106                                KEGG_TIGHT_JUNCTION  0.013798  0.109754   \n",
      "108           KEGG_COMPLEMENT_AND_COAGULATION_CASCADES  0.042629  0.204882   \n",
      "132       KEGG_PROGESTERONE_MEDIATED_OOCYTE_MATURATION  0.048254  0.211111   \n",
      "141                            KEGG_ALZHEIMERS_DISEASE  0.000266  0.008468   \n",
      "142                            KEGG_PARKINSONS_DISEASE  0.000135  0.006244   \n",
      "144                           KEGG_HUNTINGTONS_DISEASE  0.000134  0.006244   \n",
      "145                                KEGG_PRION_DISEASES  0.020822  0.145756   \n",
      "146                     KEGG_VIBRIO_CHOLERAE_INFECTION  0.025872  0.156127   \n",
      "167                  KEGG_SYSTEMIC_LUPUS_ERYTHEMATOSUS  0.044651  0.204882   \n",
      "\n",
      "           ES       NES  nMoreExtreme  size  leadingEdge  \n",
      "1    0.547111  1.717589            33    28          NaN  \n",
      "6    0.594532  1.615124            84    16          NaN  \n",
      "7    0.415302  1.398868           417    39          NaN  \n",
      "10   0.558980  1.689741            45    24          NaN  \n",
      "11   0.541345  2.116049             0    90          NaN  \n",
      "14   0.604427  1.897524             3    28          NaN  \n",
      "17   0.552392  1.875476             2    41          NaN  \n",
      "19   0.547871  1.656160            58    24          NaN  \n",
      "20   0.518489  1.705473            31    35          NaN  \n",
      "24   0.608842  1.927103             3    29          NaN  \n",
      "35  -0.750362 -1.654825            80     6          NaN  \n",
      "48   0.617985  1.478837           315    10          NaN  \n",
      "49   0.500974  1.572748           120    28          NaN  \n",
      "50   0.479204  1.490929           249    27          NaN  \n",
      "54   0.658941  1.576844           144    10          NaN  \n",
      "57   0.564159  1.687789            44    23          NaN  \n",
      "61   0.552395  1.500652           265    16          NaN  \n",
      "64   0.593996  1.777053            19    23          NaN  \n",
      "67   0.413574  1.435351           307    45          NaN  \n",
      "69   0.636515  1.699993            30    15          NaN  \n",
      "72   0.687294  1.535138           183     8          NaN  \n",
      "76  -0.310068 -1.345832           108    76          NaN  \n",
      "79  -0.281236 -1.302744            88   112          NaN  \n",
      "82  -0.432127 -1.916093             0    86          NaN  \n",
      "83   0.419709  1.413713           379    39          NaN  \n",
      "84   0.498118  1.549776           153    27          NaN  \n",
      "85   0.485798  1.548392           154    30          NaN  \n",
      "88   0.410228  1.453412           232    51          NaN  \n",
      "90   0.557240  1.946593             0    47          NaN  \n",
      "93   0.427980  1.441572           313    39          NaN  \n",
      "99  -0.518088 -1.687484            43    21          NaN  \n",
      "102 -0.451503 -1.941829             0    73          NaN  \n",
      "103 -0.572447 -2.207507             0    43          NaN  \n",
      "104 -0.378200 -1.671816             3    85          NaN  \n",
      "106 -0.401102 -1.546756            41    43          NaN  \n",
      "108 -0.365405 -1.401211           130    42          NaN  \n",
      "132  0.545288  1.481346           303    16          NaN  \n",
      "141  0.492825  1.914140             1    86          NaN  \n",
      "142  0.504846  1.936789             0    80          NaN  \n",
      "144  0.533168  2.050072             0    81          NaN  \n",
      "145 -0.587613 -1.663543            78    13          NaN  \n",
      "146  0.488622  1.533970           171    28          NaN  \n",
      "167 -0.367315 -1.403624           138    41          NaN  \n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                      Late_vs_Early_MDS_GSEA_Pathways.csv\t                      \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "                                          pathway      pval      padj  \\\n",
      "0                      KEGG_FATTY_ACID_METABOLISM  0.028030  0.056394   \n",
      "1  KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION  0.081128  0.121692   \n",
      "3                     KEGG_PPAR_SIGNALING_PATHWAY  0.002595  0.015572   \n",
      "4                                 KEGG_PEROXISOME  0.028197  0.056394   \n",
      "\n",
      "         ES       NES  nMoreExtreme  size  leadingEdge  \n",
      "0  0.756713  1.652515           107     5          NaN  \n",
      "1  0.733443  1.479070           327     4          NaN  \n",
      "3  0.860951  1.880148             9     5          NaN  \n",
      "4  0.813519  1.640554           113     4          NaN  \n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                   Late_vs_Early_MDS_TMDE_GSEA_Pathways.csv\t                    \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "Empty DataFrame\n",
      "Columns: [pathway, pval, padj, ES, NES, nMoreExtreme, size, leadingEdge]\n",
      "Index: []\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                     Late_vs_Early_TPDE_GSEA_Pathways.csv\t                      \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "Empty DataFrame\n",
      "Columns: [pathway, pval, padj, ES, NES, nMoreExtreme, size, leadingEdge]\n",
      "Index: []\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                      Late_vs_Early_MDE_GSEA_Pathways.csv\t                      \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "               pathway      pval      padj        ES       NES  nMoreExtreme  \\\n",
      "17  KEGG_AXON_GUIDANCE  0.000679  0.042748  0.813272  1.855256             3   \n",
      "\n",
      "    size  leadingEdge  \n",
      "17     8          NaN  \n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                     Late_vs_Early_TPDS_GSEA_Pathways.csv\t                      \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "                                         pathway      pval      padj  \\\n",
      "3         KEGG_ASCORBATE_AND_ALDARATE_METABOLISM  0.042114  0.237198   \n",
      "13                       KEGG_LYSINE_DEGRADATION  0.036252  0.237198   \n",
      "14          KEGG_ARGININE_AND_PROLINE_METABOLISM  0.013734  0.237198   \n",
      "18                    KEGG_TRYPTOPHAN_METABOLISM  0.002377  0.123613   \n",
      "19                  KEGG_BETA_ALANINE_METABOLISM  0.050911  0.237198   \n",
      "20              KEGG_SELENOAMINO_ACID_METABOLISM  0.033902  0.237198   \n",
      "22                  KEGG_GLYCEROLIPID_METABOLISM  0.037713  0.237198   \n",
      "25                      KEGG_PYRUVATE_METABOLISM  0.037587  0.237198   \n",
      "27                    KEGG_PROPANOATE_METABOLISM  0.054738  0.237198   \n",
      "32          KEGG_LIMONENE_AND_PINENE_DEGRADATION  0.037312  0.237198   \n",
      "44                           KEGG_TIGHT_JUNCTION  0.051334  0.237198   \n",
      "48  KEGG_PROXIMAL_TUBULE_BICARBONATE_RECLAMATION  0.019702  0.237198   \n",
      "\n",
      "          ES       NES  nMoreExtreme  size  leadingEdge  \n",
      "3   0.700539  1.532372           211     6          NaN  \n",
      "13  0.609111  1.575044           182    10          NaN  \n",
      "14  0.586664  1.690474            69    14          NaN  \n",
      "18  0.709532  1.834715            11    10          NaN  \n",
      "19  0.590893  1.527938           256    10          NaN  \n",
      "20  0.747188  1.531119           168     5          NaN  \n",
      "22  0.740437  1.517286           187     5          NaN  \n",
      "25  0.588003  1.563041           189    11          NaN  \n",
      "27  0.525767  1.515000           278    14          NaN  \n",
      "32  0.740906  1.518246           185     5          NaN  \n",
      "44 -0.567427 -1.521261           253    11          NaN  \n",
      "48 -0.753408 -1.661913            97     6          NaN  \n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                     Late_vs_Early_TMDS_GSEA_Pathways.csv\t                      \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "                                           pathway      pval      padj  \\\n",
      "1                     KEGG_CITRATE_CYCLE_TCA_CYCLE  0.007217  0.099599   \n",
      "10  KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION  0.026498  0.203154   \n",
      "40                           KEGG_ABC_TRANSPORTERS  0.013986  0.130040   \n",
      "47                                 KEGG_PEROXISOME  0.002829  0.097606   \n",
      "52               KEGG_CELL_ADHESION_MOLECULES_CAMS  0.005419  0.099599   \n",
      "53                          KEGG_ADHERENS_JUNCTION  0.001553  0.097606   \n",
      "54                             KEGG_TIGHT_JUNCTION  0.012353  0.130040   \n",
      "57           KEGG_FC_GAMMA_R_MEDIATED_PHAGOCYTOSIS  0.015077  0.130040   \n",
      "58       KEGG_LEUKOCYTE_TRANSENDOTHELIAL_MIGRATION  0.006211  0.099599   \n",
      "\n",
      "          ES       NES  nMoreExtreme  size  leadingEdge  \n",
      "1   0.592830  1.656132            62    16          NaN  \n",
      "10  0.541362  1.532047           232    17          NaN  \n",
      "40  0.822230  1.557723            99     4          NaN  \n",
      "47  0.548184  1.694294            25    26          NaN  \n",
      "52 -0.656031 -1.925157            10     8          NaN  \n",
      "53 -0.853799 -2.014262             3     5          NaN  \n",
      "54 -0.558370 -1.798850            21    10          NaN  \n",
      "57 -0.818702 -1.751089            42     4          NaN  \n",
      "58 -0.787309 -1.857400            15     5          NaN  \n"
     ]
    }
   ],
   "source": [
    "# Need to add in entrez gene ID and also make labels for genes for ORA\n",
    "# Imports\n",
    "import pandas as pd\n",
    "import os\n",
    "from sciutil import SciUtil\n",
    "\n",
    "# Setup file locations and label of the cancer\n",
    "u = SciUtil()\n",
    "\n",
    "cancer = 'ClearCellRenalCellCarcinoma'\n",
    "input_dir = 'Input_RCM'\n",
    "output_dir = 'Output_Data'\n",
    "supp_dir = 'Required_Refs'\n",
    "fig_dir = 'Output_Figures'\n",
    "regLabel = 'RG2_Changes_filtered'\n",
    "\n",
    "files = [f for f in os.listdir(fig_dir) if 'GSEA' in f]\n",
    "\n",
    "# Save consolidated pathways for each of the analyses\n",
    "conditions = ['Stage_IV_vs_Stage_I', 'Late_vs_Early']\n",
    "for cond in conditions:\n",
    "    c_df = pd.DataFrame()\n",
    "    for f in files:\n",
    "        if cond in f and '_Pathways' in f:\n",
    "            df = pd.read_csv(os.path.join(fig_dir, f))\n",
    "            c_df = pd.concat([c_df, df])\n",
    "            u.dp([f])\n",
    "            print(df[df['padj'] < 0.25])\n",
    "    c_df.to_csv(os.path.join(output_dir, f'{cond}_Pathways.tsv'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "af575039",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m              Stage_IV_vs_Stage_I_All_GSEA_MetabolicPathways.csv\t               \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                 Late_vs_Early_TMDE_GSEA_MetabolicPathways.csv\t                 \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                 Late_vs_Early_TMDS_GSEA_MetabolicPathways.csv\t                 \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                 Late_vs_Early_All_GSEA_MetabolicPathways.csv\t                  \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                 Late_vs_Early_TPDS_GSEA_MetabolicPathways.csv\t                 \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m               Late_vs_Early_MDE_TMDS_GSEA_MetabolicPathways.csv\t               \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m              Late_vs_Early_TPDE_TMDS_GSEA_MetabolicPathways.csv\t               \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                 Late_vs_Early_MDS_GSEA_MetabolicPathways.csv\t                  \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m              Late_vs_Early_TPDS_TMDE_GSEA_MetabolicPathways.csv\t               \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m               Late_vs_Early_MDS_TMDE_GSEA_MetabolicPathways.csv\t               \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                 Late_vs_Early_MDE_GSEA_MetabolicPathways.csv\t                  \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[94m                 Late_vs_Early_TPDE_GSEA_MetabolicPathways.csv\t                 \u001b[0m\n",
      "\u001b[94m--------------------------------------------------------------------------------\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# Save consolidated pathways for each of the analyses\n",
    "conditions = ['Stage_IV_vs_Stage_I', 'Late_vs_Early']\n",
    "for cond in conditions:\n",
    "    c_df = pd.DataFrame()\n",
    "    for f in files:\n",
    "        if cond in f and 'MetabolicPathways' in f:\n",
    "            df = pd.read_csv(os.path.join(fig_dir, f))\n",
    "            c_df = pd.concat([c_df, df])\n",
    "            u.dp([f])\n",
    "            #print(df[df['padj'] < 0.25])\n",
    "    c_df.to_csv(os.path.join(output_dir, f'{cond}_MetabolicPathways.tsv'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad0e5893",
   "metadata": {},
   "source": [
    "# Now look at what the pathways are for each of the S1 and S4/Early and late"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "db70c154",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'data_dir' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 41\u001b[0m\n\u001b[1;32m     39\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m c \u001b[38;5;129;01min\u001b[39;00m comparisons:\n\u001b[1;32m     40\u001b[0m     combined_df \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame()\n\u001b[0;32m---> 41\u001b[0m     rna_df \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mread_csv(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[43mdata_dir\u001b[49m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;132;01m{\u001b[39;00mcomparisons[c][\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mreplace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIV\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;250m \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mI\u001b[39m\u001b[38;5;124m\"\u001b[39m)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m     42\u001b[0m     rna_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGeneRatio\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m [\u001b[38;5;28mint\u001b[39m(g\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m/\u001b[39m\u001b[38;5;124m'\u001b[39m)[\u001b[38;5;241m0\u001b[39m])\u001b[38;5;241m/\u001b[39m\u001b[38;5;28mint\u001b[39m(g\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m/\u001b[39m\u001b[38;5;124m'\u001b[39m)[\u001b[38;5;241m1\u001b[39m]) \u001b[38;5;28;01mfor\u001b[39;00m g \u001b[38;5;129;01min\u001b[39;00m rna_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGeneRatio\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mvalues]\n\u001b[1;32m     43\u001b[0m     rna_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mData\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mRNA S1\u001b[39m\u001b[38;5;124m'\u001b[39m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'data_dir' is not defined"
     ]
    }
   ],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib as mpl\n",
    "import pandas as pd\n",
    "import matplotlib.cm as cm\n",
    "import numpy as np\n",
    "import matplotlib.cm as cm\n",
    "import matplotlib.pyplot as plt\n",
    "plt.rcParams[\"figure.figsize\"] = (3,10)\n",
    "\n",
    "sns.set_theme(style=\"whitegrid\")\n",
    "#cmaps = {'Down regulated': {'CpG': 'darkorange', 'Protein': 'royalblue', 'RNA': 'teal'},\n",
    "#         'Up regulated': {'CpG': 'limegreen', 'Protein': 'darkred', 'RNA': 'lightcoral'},\n",
    "#         }\n",
    "cmaps = {'Down regulated': {'CpG S1': 'darkorange', 'Protein S1': 'royalblue', 'RNA S1': 'teal', \n",
    "                           'CpG S4': 'darkorange', 'Protein S4': 'royalblue', 'RNA S4': 'teal'},\n",
    "        'Up regulated': {'CpG S1': 'limegreen', 'Protein S1': 'darkred', 'RNA S1': 'lightcoral', \n",
    "                           'CpG S4': 'limegreen', 'Protein S4': 'darkred', 'RNA S4': 'lightcoral'},\n",
    "        }\n",
    "\n",
    "# cmaps = {'Down regulated': {'CpG S1': 'chocolate', 'Protein S1': 'royalblue', 'RNA S1': 'teal', \n",
    "#                            'CpG S4': 'orange', 'Protein S4': 'navy', 'RNA S4': 'turquoise'},\n",
    "#         'Up regulated': {'CpG S1': 'green', 'Protein S1': 'red', 'RNA S1': 'orange', \n",
    "#                            'CpG S4': 'limegreen', 'Protein S4': 'darkred', 'RNA S4': 'lightcoral'},\n",
    "#         }\n",
    "\n",
    "comparisons = {'Down regulated': ['SiRCle_StageIV__RNAseq_Negative_LogFC-1_pvalue0.05.csv',\n",
    "                                  'SiRCle_StageIV__Protein_Negative_LogFC-1_pvalue0.05.csv',\n",
    "                                  'SiRCle_StageI__Methylation_Positive_LogFC0.1_pvalue0.05.csv'\n",
    "                                 ],\n",
    "               'Up regulated': ['SiRCle_StageIV__RNAseq_Positive_LogFC1_pvalue0.05.csv',\n",
    "                                'SiRCle_StageIV__Protein_Positive_LogFC1_pvalue0.05.csv',\n",
    "                                 'SiRCle_StageIV__Methylation_Negative_LogFC-0.1_pvalue0.05.csv'\n",
    "                                 ]\n",
    "              }\n",
    "\n",
    "num = 10\n",
    "sizes = {'Down regulated': {'min': 18, 'max': 374},\n",
    "        'Up regulated': {'min': 16, 'max': 448}}\n",
    "for c in comparisons:\n",
    "    combined_df = pd.DataFrame()\n",
    "    rna_df = pd.read_csv(f'{data_dir}{comparisons[c][0].replace(\"IV\", \"I\")}')\n",
    "    rna_df['GeneRatio'] = [int(g.split('/')[0])/int(g.split('/')[1]) for g in rna_df['GeneRatio'].values]\n",
    "    rna_df['Data'] = 'RNA S1'\n",
    "    rna_df.sort_values('p.adjust', inplace=True)\n",
    "\n",
    "    protein_df = pd.read_csv(f'{data_dir}{comparisons[c][1].replace(\"IV\", \"I\")}')\n",
    "    protein_df['GeneRatio'] = [int(g.split('/')[0])/int(g.split('/')[1]) for g in protein_df['GeneRatio'].values]\n",
    "    protein_df['Data'] = 'Protein S1'\n",
    "    protein_df.sort_values('p.adjust', inplace=True)\n",
    "\n",
    "    cpg_df = pd.read_csv(f'{data_dir}{comparisons[c][2].replace(\"IV\", \"I\")}')\n",
    "    cpg_df['GeneRatio'] = [int(g.split('/')[0])/int(g.split('/')[1]) for g in cpg_df['GeneRatio'].values]\n",
    "    cpg_df['Data'] = 'CpG S1'\n",
    "    cpg_df.sort_values('p.adjust', inplace=True)\n",
    "    combined_df = combined_df.append(rna_df)\n",
    "    combined_df = combined_df.append(cpg_df)\n",
    "    combined_df = combined_df.append(protein_df)\n",
    "    \n",
    "    rna_df = pd.concat([rna_df.head(num), cpg_df.head(num), protein_df.head(num)])\n",
    "    rna_df.sort_values('p.adjust', inplace=True)\n",
    "\n",
    "    max_c = sizes[c]['max'] ##max(rna_df['Count'].values)\n",
    "    min_c = sizes[c]['min']  #min(rna_df['Count'].values)\n",
    "    size = [int(1 + ((int(c) - min_c)/(max_c - min_c))*300) for c in rna_df['Count'].values]\n",
    "    \n",
    "\n",
    "    norm = mpl.colors.Normalize(vmin=np.min(rna_df['p.adjust'].values), vmax=np.max(rna_df['p.adjust'].values))\n",
    "    cmap = cm.RdBu\n",
    "    m = cm.ScalarMappable(norm=norm, cmap=cmap)\n",
    "    c_map = cmaps[c]\n",
    "    colours = [c_map.get(v) for v in rna_df['Data'].values]\n",
    "    plt.scatter(rna_df['GeneRatio'].values, rna_df['Description'].values, s=size, c=colours, marker=',', \n",
    "                edgecolors='black', alpha=0.8)#c=m.to_rgba(rna_df['p.adjust'].values))\n",
    "\n",
    "        \n",
    "    rna_df = pd.read_csv(f'{data_dir}{comparisons[c][0]}')\n",
    "    rna_df['GeneRatio'] = [int(g.split('/')[0])/int(g.split('/')[1]) for g in rna_df['GeneRatio'].values]\n",
    "    rna_df['Data'] = 'RNA S4'\n",
    "    rna_df.sort_values('p.adjust', inplace=True)\n",
    "\n",
    "    protein_df = pd.read_csv(f'{data_dir}{comparisons[c][1]}')\n",
    "    protein_df['GeneRatio'] = [int(g.split('/')[0])/int(g.split('/')[1]) for g in protein_df['GeneRatio'].values]\n",
    "    protein_df['Data'] = 'Protein S4'\n",
    "    protein_df.sort_values('p.adjust', inplace=True)\n",
    "\n",
    "    cpg_df = pd.read_csv(f'{data_dir}{comparisons[c][2]}')\n",
    "    cpg_df['GeneRatio'] = [int(g.split('/')[0])/int(g.split('/')[1]) for g in cpg_df['GeneRatio'].values]\n",
    "    cpg_df['Data'] = 'CpG S4'\n",
    "    cpg_df.sort_values('p.adjust', inplace=True)\n",
    "    combined_df = combined_df.append(rna_df)\n",
    "    combined_df = combined_df.append(cpg_df)\n",
    "    combined_df = combined_df.append(protein_df)\n",
    "    # Concat all together\n",
    "    num = 10\n",
    "    rna_df = pd.concat([rna_df.head(num), cpg_df.head(num), protein_df.head(num)])\n",
    "    rna_df.sort_values('p.adjust', inplace=True)\n",
    "\n",
    "    max_c = sizes[c]['max'] ##max(rna_df['Count'].values)\n",
    "    min_c = sizes[c]['min']  #min(rna_df['Count'].values)\n",
    "    size = [int(1 + ((int(c) - min_c)/(max_c - min_c))*300) for c in rna_df['Count'].values]\n",
    "\n",
    "\n",
    "    norm = mpl.colors.Normalize(vmin=np.min(rna_df['p.adjust'].values), vmax=np.max(rna_df['p.adjust'].values))\n",
    "    cmap = cm.RdBu\n",
    "    m = cm.ScalarMappable(norm=norm, cmap=cmap)\n",
    "    c_map = cmaps[c]\n",
    "    colours = [c_map.get(v) for v in rna_df['Data'].values]\n",
    "    plt.scatter(rna_df['GeneRatio'].values, rna_df['Description'].values, s=size, c=colours, edgecolors='black', marker='*',  alpha=0.8)#c=m.to_rgba(rna_df['p.adjust'].values))\n",
    "    plt.title(c)\n",
    "    \n",
    "#     # Add legends\n",
    "#     cmap_labels = [f'G.shared:{min(overlapping_numbers)}-{max(overlapping_numbers)}']\n",
    "#     # create proxy artists as handles:\n",
    "#     cmaps = [edge_cmap]\n",
    "#     cmap_handles = [Rectangle((0, 0), 1, 1) for _ in cmaps]\n",
    "#     handler_map = dict(zip(cmap_handles,\n",
    "#                            [HandlerColormap(cm, num_stripes=8) for cm in cmaps]))\n",
    "#     legend2 = plt.legend(handles=cmap_handles,\n",
    "#                labels=cmap_labels,\n",
    "#                handler_map=handler_map,\n",
    "#                fontsize=8, loc='upper right')\n",
    "\n",
    "#     plt.gca().add_artist(legend2)\n",
    "    combined_df.to_csv(f'{output_dir}{c}_StageIV-StageI-TvN.csv', index=False)\n",
    "\n",
    "    gmin = plt.scatter([], [], s=10 + int(300*(min_c/(max_c - min_c))), marker='o', color='#222')\n",
    "    #gmid = plt.scatter([], [], s=int(10 + 300*(((max_c - min_c)/2)/max_c)), marker='o', color='#222')\n",
    "    gmax = plt.scatter([], [], s=10 + int(300*(1)), marker='o', color='#222')\n",
    "    legend = plt.legend((gmin, gmax),\n",
    "                        (str(min_c), str(max_c)),\n",
    "                        scatterpoints=1,\n",
    "                        loc='lower left',\n",
    "                        ncol=1,\n",
    "                        fontsize=8, bbox_to_anchor=(0, -0.1))\n",
    "    \n",
    "    legend.set_title(\"No. Genes\")\n",
    "    plt.gca().add_artist(legend)\n",
    "    plt.savefig(f'{output_dir}DotPlot_GO_{c}_S1-S4.svg')\n",
    "    plt.show()\n",
    "#sns.stripplot(data=rna_df.head(30), y='Description', x='GeneRatio', size=size) #, color='p.adjust', size='Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "26f89f8c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
