

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>References &mdash; SiRCle_multiomics_integration 0.0.1 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />

  
  
    <link rel="shortcut icon" href="_static/logo.png"/>
  

  
  

  
    <link rel="canonical" href="https://github.com/ArianeMora/SiRCle_multiomics_integration/references.html" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script src="_static/custom.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

    <link rel="author" title="About these documents" href="about.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Change log" href="changelog.html" />
 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html">
          

          
            
            <img src="_static/logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                0.0.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">Package info</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="about.html">SUMMARY</a></li>
<li class="toctree-l1"><a class="reference internal" href="about.html#abstract">Abstract</a></li>
<li class="toctree-l1"><a class="reference internal" href="installing/index.html">Installing SiRCle</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Reproducibility</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="examples/N01_RNAProcessing.html">Notebook for RNA processing part 1</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N02_ClinicalProcessing.html">Notebook for Clinical processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N03_protein_imputation.html">Protein Imputation</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N04_ProteinProcessing.html">Notebook for Protein processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N05_MethProcessing.html">Notebook for Methylation processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N06_PhosphoProteinProcessing.html">Notebook for Phospho-proteoimcs peptide processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N07_RNAProcessing.html">Notebook for RNA processing part 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N08_DatasetGeneration.html">Notebook for Generating datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N09_TvN_DE_DA_CpG.html">Tumour vs normal comparison</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N10_MethylationFilter.html">Notebook for Filtering CpGs to Genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N11_SiRCle_RCM.html">Notebook for performing SiRCle clustering</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N12_SiRCle_ORA_GSEA.html">ORA for SiRCle clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N13_SiRCle_ORA_vis.html">Notebook for Over Representation Analysis Visualisation for SiRCle</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N14_TF.html">Notebook for TF analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N15_VAE_Integration.html">Notebook for VAE integration</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N16_Integration_GSEA.html">GSEA on the integrated VAE value</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N17_SingleCell_setup.html">Set up single cell files</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N17_singlecell_StageIV-StageI.html">Notebook for Single cell analysis using integrated genes Stage IV vs Stage I</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N18_singlecell_PBRM1-BAP1.html">Notebook for Single cell analysis using integrated genes for PBRM1 vs BAP1</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N20_Metabolomics_TvN.html">Notebook for metabolomics analysis using publicly available data</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N21_Metabolomics_RCM.html">Notebook for RCM data with metabolic pathways</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/N22_Metabolomics_VAE.html">Notebook for VAE data with metabolic pathways</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/Benchmarking.html">Notebook for MOMIX benchmarking</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="changelog.html">Change log</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">References</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#code">Code</a></li>
<li class="toctree-l2"><a class="reference internal" href="#literature">Literature</a></li>
<li class="toctree-l2"><a class="reference internal" href="#blogs-and-githubs-used">Blogs and Githubs used</a></li>
<li class="toctree-l2"><a class="reference internal" href="#data">Data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#environments">Environments</a></li>
</ul>
</li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">SiRCle_multiomics_integration</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content style-external-links">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>References</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            
              <a href="https://github.com/arianemora/SiRCle_multiomics_integration/blob/main/docs_src/src/references.rst" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="references">
<span id="id1"></span><h1>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h1>
<div class="section" id="code">
<h2>Code<a class="headerlink" href="#code" title="Permalink to this headline">¶</a></h2>
<p>sci-vae is designed for Python &gt;=3.6 and requires the following libraries, which will be automatically installed:</p>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 19%" />
<col style="width: 19%" />
<col style="width: 63%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Library</p></th>
<th class="head"><p>Version</p></th>
<th class="head"><p>Reference</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://keras.io/">keras</a></p></td>
<td><p>&gt;= 2.4.0</p></td>
<td><p>&#64;misc{chollet2015keras, title={Keras}, author={Chollet, Franc{c}ois and others}, year={2015}, howpublished={url{<a class="reference external" href="https://keras.io">https://keras.io</a>}},}</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.tensorflow.org/">Tensorflow</a> and <a class="reference external" href="https://www.tensorflow.org/probability/">tensorflow-probability</a></p></td>
<td><p>&gt;= 2.1.0</p></td>
<td><p>Abadi, M., Barham, P., Chen, J., Chen, Z., Davis, A., Dean, J., Devin, M., Ghemawat, S., Irving, G., Isard, M., Kudlur, M., Levenberg, J., Monga, R., Moore, S., Murray, D. G., Steiner, B., Tucker, P., Vasudevan, V., Warden, P., … Zheng, X. (2016). TensorFlow: A system for large-scale machine learning. 12th USENIX Symposium on Operating Systems Design and Implementation (OSDI 16), 265–283. <a class="reference external" href="https://www.usenix.org/system/files/conference/osdi16/osdi16-abadi.pdf">https://www.usenix.org/system/files/conference/osdi16/osdi16-abadi.pdf</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://numpy.org/">NumPy</a></p></td>
<td><p>&gt;= 1.9.0</p></td>
<td><p>Harris, C.R., Millman, K.J., van der Walt, S.J. et al. Array programming with NumPy. Nature 585, 357–362 (2020). DOI: <a class="reference external" href="https://doi.org/10.1038/s41586-020-2649-2">0.1038/s41586-020-2649-2</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://pandas.pydata.org/">pandas</a></p></td>
<td><p>&gt;= 1.0.3</p></td>
<td><a class="reference external image-reference" href="https://doi.org/10.5281/zenodo.3715232"><img alt="https://zenodo.org/badge/DOI/10.5281/zenodo.3715232.svg" src="https://zenodo.org/badge/DOI/10.5281/zenodo.3715232.svg" /></a>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://scikit-learn.org/stable/about.html">scikit-learn</a></p></td>
<td><p>&gt;= 0.22.2</p></td>
<td><p>Pedregosa, F., Varoquaux, G., Gramfort, A., Michel, V., Thirion, B., Grisel, O., Blondel, M., Prettenhofer, P., Weiss, R., Dubourg, V., Vanderplas, J., Passos, A., Cournapeau, D., Brucher, M., Perrot, M., &amp; Duchesnay, É. (2011). Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research, 12(85), 2825–2830.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://pythonhosted.org/pyswarm/">pyswarm</a></p></td>
<td><p>&gt;= 0.6</p></td>
<td><p><a class="reference external" href="https://github.com/tisimst/pyswarm">https://github.com/tisimst/pyswarm</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://matplotlib.org/3.3.3/">matplotlib</a></p></td>
<td><p>&gt;= 3.3.3</p></td>
<td><a class="reference external image-reference" href="https://doi.org/10.5281/zenodo.3264781"><img alt="https://zenodo.org/badge/DOI/10.5281/zenodo.3264781.svg" src="https://zenodo.org/badge/DOI/10.5281/zenodo.3264781.svg" /></a>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://seaborn.pydata.org/">seaborn</a></p></td>
<td><p>&gt;= 0.11.1</p></td>
<td><a class="reference external image-reference" href="https://doi.org/10.5281/zenodo.4379347"><img alt="https://zenodo.org/badge/DOI/10.5281/zenodo.4379347.svg" src="https://zenodo.org/badge/DOI/10.5281/zenodo.4379347.svg" /></a>
</td>
</tr>
</tbody>
</table>
<p><strong>We strongly encourage you to cite the other python packages</strong>. Note <em>pyswarm</em> and <em>scikit-learn</em> are only used when
using the <em>optimiser</em> component (not your standard VAE) so these can be omitted if you don’t use them. Similarly,
matplotlib and seaborn may not necessarily be used, so it depends on what aspects of this project is used in your work.</p>
</div>
<div class="section" id="literature">
<h2>Literature<a class="headerlink" href="#literature" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 23%" />
<col style="width: 77%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Label</p></th>
<th class="head"><p>Reference</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>VAE paper</p></td>
<td><p>Kingma, Diederik P, and Max Welling. 2014. “Auto-Encoding Variational Bayes.” In International Conference on Learning Representations.</p></td>
</tr>
<tr class="row-odd"><td><p>Info VAE</p></td>
<td><p>Zhao, S., Song, J., &amp; Ermon, S. (2018). InfoVAE: Information Maximizing Variational Autoencoders. ArXiv:1706.02262 [Cs, Stat]. http://arxiv.org/abs/1706.02262</p></td>
</tr>
<tr class="row-even"><td><p>VAEs applied to cancer</p></td>
<td><p>Simidjievski, N., Bodnar, C., Tariq, I., Scherer, P., Andres Terre, H., Shams, Z., Jamnik, M., &amp; Liò, P. (2019). Variational Autoencoders for Cancer Data Integration: Design Principles and Computational Practice. Frontiers in Genetics, 10. <a class="reference external" href="https://doi.org/10.3389/fgene.2019.01205">https://doi.org/10.3389/fgene.2019.01205</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="blogs-and-githubs-used">
<h2>Blogs and Githubs used<a class="headerlink" href="#blogs-and-githubs-used" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 100%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Link</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://github.com/pren1/keras-MMD-Variational-Autoencoder/blob/master/Keras_MMD_Variational_Autoencoder.ipynb">MMD in Keras</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/s-omranpour/X-VAE-keras/blob/master/VAE/VAE_MMD.ipynb">X-VAE Keras</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://github.com/ShengjiaZhao/MMD-Variational-Autoencoder">MMD VAE in Tensorflow</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/s-omranpour/X-VAE-keras/blob/master/VAE/VAE_MMD.ipynb">X-VAE Keras</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://github.com/CancerAI-CL/IntegrativeVAEs/blob/master/code/models/mmvae.py">VAE applied to cancer data</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://ermongroup.github.io/blog/a-tutorial-on-mmd-variational-autoencoders/">Blog by Ermon group re. InfoVAE</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="data">
<h2>Data<a class="headerlink" href="#data" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-given docutils align-default">
<colgroup>
<col style="width: 19%" />
<col style="width: 19%" />
<col style="width: 63%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Link</p></th>
<th class="head"><p>Reference</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0092867419311237">CPTAC ccRCC paper</a></p></td>
<td><p><a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0092867419311237">https://www.sciencedirect.com/science/article/pii/S0092867419311237</a></p></td>
<td><p>Clark, David J. et al. 2019. ‘Integrated Proteogenomic Characterization of Clear Cell Renal Cell Carcinoma’. Cell 179(4): 964-983.e31.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S1535610821001653?via%3Dihub">Single cell ccRCC data</a></p></td>
<td><p><a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S1535610821001653?via%3Dihub">https://www.sciencedirect.com/science/article/pii/S1535610821001653?via%3Dihub</a></p></td>
<td><p>Krishna, Chirag et al. 2021. ‘Single-Cell Sequencing Links Multiregional Immune Landscapes and Tissue-Resident T Cells in CcRCC to Tumor Topology and Therapy Efficacy’. Cancer Cell 39(5): 662-677.e6.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://www.nature.com/articles/s41585-019-0211-5">TCGA methylation data</a></p></td>
<td><p><a class="reference external" href="https://www.nature.com/articles/s41585-019-0211-5">https://www.nature.com/articles/s41585-019-0211-5</a></p></td>
<td><p>Linehan, W. M. &amp; Ricketts, C. J. The Cancer Genome Atlas of renal cell carcinoma: findings and clinical implications. Nature Reviews Urology 16, 539–552 (2019).</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="environments">
<h2>Environments<a class="headerlink" href="#environments" title="Permalink to this headline">¶</a></h2>
<p>This will be user dependent but don’t forget to cite anaconda if you use it or <a class="reference external" href="https://rstudio.github.io/reticulate/">reticulate</a> if you use R.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="changelog.html" class="btn btn-neutral float-left" title="Change log" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2022, Ariane Mora and Christina Schmidt.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>